### PR TITLE
fix long sfp chan names

### LINKED ISSUE #3385: .sfp import with long ch names
The .sfp chan loc importer currently truncates long channel names. I came across this in a file with eye tracker channels included - 'L_GAZE_X' and 'L_GAZE_Y', which were truncated to have the same name, leading to all kinds of problems.

Could we simply do `dtype = np.dtype('S16, f8, f8, f8')` instead of the `dtype = np.dtype('S4, f8, f8, f8')` we currently have? 

This would still break at very long names. Possibly, a safer writer would process the strings line by line. But maybe long ch names are so rare it's not worth it.

### PR DESCRIPTION
**Pull request by jona-sassenhagen**
Replace file parser for .sfp files. Less efficient, but should not choke on long channel names.

~~Missing dedicated test for the specific scenario this is aimed to fix.~~

Closes #3385

### CODE REVIEW

File: mne/channels/montage.py [hunk 1]
```diff
@@ -187,10 +187,17 @@ def read_montage(kind, ch_names=None, path=None, unit='m', transform=False):
...
-        dtype = np.dtype('S4, f8, f8, f8')
-        data = np.loadtxt(fname, dtype=dtype)
-        pos = np.c_[data['f1'], data['f2'], data['f3']]
-        ch_names_ = data['f0'].astype(np.str)
+        with open(fname, 'r') as f:
+            lines = f.read().replace('\t', ' ').split("\n")
+
+        ch_names_, pos = [], []
+        for line in lines:
+            line = line.lstrip().rstrip().split()
```
Comment (agramfort) [inline mne/channels/montage.py]: just line.strip ?

File: mne/channels/montage.py [hunk 2]
```diff
@@ -187,10 +187,17 @@ def read_montage(kind, ch_names=None, path=None, unit='m', transform=False):
...
+            if len(line) > 0:  # skip empty lines
+                name, x, y, z = line
+                ch_names_.append(name)
+                pos.append([np.float(cord) for cord in (x, y, z)])
```
Comment (agramfort) [inline mne/channels/montage.py]: pos.append(map(float, (x, y, z)))

should be enough

Comment (agramfort) [inline mne/channels/montage.py]: pos.append(map(float, (x, y, z)))

should be enough

Comment (jona-sassenhagen) [inline mne/channels/montage.py]: I thought list comp is more pythonic than map?

http://docs.quantifiedcode.com/python-anti-patterns/readability/using_map_or_filter_where_list_comprehension_is_possible.html

Comment (jona-sassenhagen) [inline mne/channels/montage.py]: Could also do:
```
                name, *coords = line
                ch_names_.append(name)
                pos.append([np.float(cord) for cord in coords])
```

Though maybe that's too clever.

I'll add an error for lines with a different number of entries than 0 or 4 though.

File: mne/channels/tests/test_montage.py [hunk 3]
```diff
@@ -73,7 +73,7 @@ def test_montage():
     """
     eeg Fp1 -95.0 -31.0 -3.0
     eeg AF7 -81 -59 -3
-    eeg AF3 -87 -41 28
+    eeg_very_very_long_name AF3 -87 -41 28
```
Comment (agramfort) [inline mne/channels/tests/test_montage.py]: you don't seem to check that the ch_name is read correctly

File: mne/channels/montage.py [hunk 4]
```diff
@@ -187,10 +187,19 @@ def read_montage(kind, ch_names=None, path=None, unit='m', transform=False):
...
+
+        ch_names_, pos = [], []
+        for ii, line in enumerate(lines):
+            line = line.strip().split()
+            if len(line) > 0:  # skip empty lines
+                if len(line) != 4:  # name, x, y, z
+                    raise ValueError("Malformed .sfp file in line " + str(ii))
+                name, *coords = line
+                ch_names_.append(name)
+                pos.append([np.float(cord) for cord in coords])
```
Comment (agramfort) [inline mne/channels/montage.py]: at least np.float -> float

**Comment by agramfort** [comment]
LGTM

let's merge when CIs are happy

**Comment by agramfort** [comment]
thx @jona-sassenhagen

**Comment by jona-sassenhagen** [comment]
Thanks Alex! Sadly, this was just the beginning - we found a few more bugs in that dataset.

**Comment by agramfort** [comment]
arfff next PR ...

**Pull request merged**